packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -178,23 +187,94 @@ export const generateValueSpecificationForParameter = (
   );
 };
 
-const fillDerivedPropertyArguments = (
+export const fillDerivedPropertyArguments = (
   derivedPropertyExpressionState: QueryBuilderDerivedPropertyExpressionState,
-  queryBuilderState: QueryBuilderState,
+  graph: PureModel,
 ): void => {
-  const propertyArguments: ValueSpecification[] =
+  let propertyArguments: ValueSpecification[] =
```

Comment:
As mentioned in the comment about `fillMilestonedDerivedPropertyArguments`, let's try to better isolate this logic.